src/zeroD/ReactorNet.cpp

```diff
@@ -305,6 +306,72 @@ void ReactorNet::addReactor(Reactor& r)
         m_integ->setMethod(BDF_Method);
         m_integ->setLinearSolverType("DENSE");
     }
+    updateNames(r);
+}
+
+void ReactorNet::updateNames(Reactor& r)
+{
+    // ensure that reactors and components have reproducible names
+    auto rType = r.type();
+    r.setDefaultName(rType, m_count[rType]);
+    m_count[rType]++;
+
+    for (size_t i=0; i<r.nWalls(); i++) {
+        auto& w = r.wall(i);
+        auto wType = w.type();
+        if (w.setDefaultName(wType, m_count[wType])) {
+            m_count[wType]++;
+        }
+        if (w.left().type() == "Reservoir") {
+            auto& res = w.left();
+            auto resType = res.type();
+            res.setDefaultName(resType, m_count[resType]);
+            m_count[resType]++;
+        }
+        if (w.right().type() == "Reservoir") {
+            auto& res = w.right();
+            auto resType = res.type();
+            res.setDefaultName(resType, m_count[resType]);
+            m_count[resType]++;
+        }
+    }
+
+    for (size_t i=0; i<r.nInlets(); i++) {
+        auto& in = r.inlet(i);
+        auto inType = in.type();
+        if (in.setDefaultName(inType, m_count[inType])) {
+            m_count[inType]++;
+        }
+        if (in.in().type() == "Reservoir") {
+            auto& res = in.in();
+            auto resType = res.type();
+            res.setDefaultName(resType, m_count[resType]);
+            m_count[resType]++;
+        }
```

Comment:
One possibility for simplifying this somewhat repetitious pattern would be to have `setDefaultName` take the `m_count` map as the one and only argument, and be responsible for updating the count for that reactor/wall/etc. type.

Reply:
:tada: Great suggestion!